File: packages/pinstripe/lib/services/session.js

export default async ({ cookies, database }) => {
    const { pinstripeSession } = cookies;
    if(!pinstripeSession){
        return;
    }
    const [ sessionId, passString ] = pinstripeSession.split(/:/);
    const session = await database.sessions.idEq(sessionId).passStringEq(passString).first();
    if(session && session.lastAccessedAt < (Date.now() - 1000 * 60 * 5)){
        await session.update({
            lastAccessedAt: Date.now()
        });
    }
    return session;
};

File: projects/slick/lib/slick/resource_factories/session.rb

require "slick/resource_factory"

Slick::ResourceFactory.register("session").define_method "create" do
    slick_session = request.cookies['slick_session']
    return nil if slick_session.to_s == ''
    (session_id, pass_string) = slick_session.split(/:/)
    sessions.id_eq(session_id).pass_string_eq(pass_string).first
end
